src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml

```diff
@@ -1477,6 +1480,42 @@ let check_required_protocol_states t ~protocol_states =
   let%map () = check_length protocol_states_assoc in
   protocol_states_assoc
 
+(* NOTE: This must be kept in sync with [Stable.Latest.hash] above! *)
+let hash_by_reading_all_proofs_from_disk_incrementally (t : t) =
+  let state_hash =
+    Parallel_scan.State.hash t.scan_state
```

Comment:
Is there a reason to have two almost identical functions instead of defining a single `hash_generic` function which takes some `~hash_node` and `~hash_leaf` parameters?

P.S. I did something very similar in #18191 and it didn't pose an issue. I'd prefer to have this approach than the current `NOTE`.

Reply:
The types are different